src/components/cc-product-list/products-controller.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * @typedef {import('./cc-product-list.types.js').CategoryData} CategoryData
+ * @typedef {import('./cc-product-list').CcProductList} CcProductList
+ * @typedef {import('./cc-product-list.types.js').Product} Product
+ * @typedef {import('./cc-product-list.types.js').CategoryFilter} CategoryFilter
+ */
+
+export class ProductsController {
+
+  /**
+   * @param {CcProductList} host
+   */
+  constructor (host) {
+    /** @type {CcProductList} The host. */
+    this._host = host;
+
+    /** @type {CategoryData[]} The initial category data list. */
+    this._categoryDataList = [];
```

Comment:
suggestion: I would go for a similar naming as the component, like @pdesoyres-cc suggested : `productsByCategories`